docs/specs/discovery.md

```diff
@@ -8,16 +8,65 @@ of a physical system:
 target device. This is essentially information intrinsic to a
-device.
+device and what capabilities it provides.
 
-These two stages occur hand-in-hand, but are distinct constructs that
-can be executed independently (i.e., there can be _discovery_ without
-_enumeration_, and _enumeration_ without _discovery_).
+These two stages occur hand-in-hand, but can be executed independently
+(i.e., there can be _discovery_ without _enumeration_,
+and _enumeration_ without _discovery_).
 
 ## Discovery
 
+A [discovery sequence](sequences/discovery.md) links various on-prem subsystems
+together for a particular device, e.g.:
+
+* MAC Address (_82:CC:18:9A:45:1C_): Low-level hardware identifier for network connection
+* IPv4 Address (_10.27.38.123_): Assigned IPv4 device network address
+* IPv6 Address (_fe80::8e8c:bc72_): Assigned IPv6 device network address
+* BACnet Address (_92EA09_): The device's BACnet mac-address
+* IoT Identifier (_AHU-32_): Device designation as used in cloud-native processing
+* Hostname (_ahu-2.acme.com_): DNS hostname for a device
+
+For any particular [discovery message](../../tests/event_discovery.tests/example.json),
+a subset of availabile information will be present (e.g. only linking the _MAC_ with
+_IPv4_ address), and it is up to the back-end systems to properly link/infer complete
+relationships. Some systems may only care about singular entries (e.g. just discovering
+what IoT devices are there, but not caring about any association).
+
+Discovery is a process that can be explicitly requested through UDMI for on-prem
+devices that support the capability (e.g. an [IoT Gateway](gateway.md)), or it
+can be done implicitly by the device itself (e.g. on a predefined interval).
+
 ## Enumeration
+
+An [enumeration sequence](sequences/enumeration.md) triggers a listing of _all_
+the parameters available from a device, rather than just the ones in its designated
+reporting set. This includes `points` (data telemetry), `blobs` (configuration elements),
+and any other relevant subcomponents. Each entry in an
+[enumeration message](../../tests/event_enumeration.tests/example.json) will include all
+the necessary identiyfing information:
+
+* `points`
+  * _point name_: The name of the point itself within the device.
+    * `units`: A list of the units that the point supports.
+    * `ref`: A reference to the point target (used to support gateway proxy functionality).
+    * `description`: Human-readable description of what the point does.
+* `blobs`
+  * _blob name_:
+    * `firmware_set`:
+    * `description`:
+
+Enumeration messages may come directly from a (direct) device itself, or
+be proxied through an [IoT Gateway](gateway.md) in a manner similar to discovery. If
+necessary, enumeration can utilize the multi-part message capability to handle a
+large number of potential points per device. The device IoT ID is included in
+the [message envelope](../messages/envelope.md).
```

Comment:
Yes, it's possible, just not clearly specified (as in, no examples). Really, the enumeration message doesn't have any sense of ID in itself, it's relying on the envelope for that.  So, the "right" solution to this might be to have conventions in the envelope for "provider" type of something, so by default the deviceId is the Cloud IoT Core ID, but it could be something else.  Or (maybe better) have a new field that can be added in that is the "raw ID" of the discovery, which can then be consumed by systems that are aware of it (and flat-out ignored by other systems). For any particular workflow I'd just want to work through the specifics with somebody in-the-know about it. The real question is that at *some* point, the IoT ID needs to be assigned... and so when/where does that happen in the overall process?